src/main/java/org/broadinstitute/hellbender/utils/mcmc/SliceSampler.java

```diff
@@ -68,23 +68,27 @@ public SliceSampler(final RandomGenerator rng, final Function<Double, Double> lo
      * @return              sample drawn from the probability density function
      */
     public double sample(final double xInitial) {
-        final double xSample = Math.min(Math.max(xInitial, xMin + EPSILON), xMax - EPSILON);
+        Utils.validateArg(xMin < xInitial && xInitial < xMax, "Initial point in slice sampler is not within specified range.");
+
         //randomly pick height of slice from uniform distribution under PDF
         //(equivalently, from exponential distribution under logPDF)
-        final double logSliceHeight = logPDF.apply(xSample) - exponentialDistribution.sample();
+        final double logSliceHeight = logPDF.apply(xInitial) - exponentialDistribution.sample();
 
         //randomly position slice with given width so that it brackets xSample; position is uniformly distributed
-        double xLeft = xSample - width * rng.nextDouble();
+        double xLeft = xInitial - width * rng.nextDouble();
         double xRight = xLeft + width;
 
         int k = MAXIMUM_NUMBER_OF_DOUBLINGS;
         //expand slice by doubling until it brackets logPDF
-        while (k > 0 && (logPDF.apply(xLeft) > logSliceHeight || logPDF.apply(xRight) > logSliceHeight) &&
-                xLeft > xMin && xRight < xMax) {
+        double logPDFLeft = xLeft > xMin ? logPDF.apply(xLeft) : Double.NEGATIVE_INFINITY;
+        double logPDFRight = xRight < xMax ? logPDF.apply(xRight) : Double.NEGATIVE_INFINITY;
+        while (k > 0 && ((logSliceHeight < logPDFLeft || logSliceHeight < logPDFRight))) {
```

Comment:
IMO the idiom of `k = MAXIMUM. . . while (k > 0 . . .) . . . k--` is less clear than `count = 0 . . . while (count < MAXIMUM . . .) . . . count++`.